mevislab.github.io/content/tutorials/thirdparty/thirdpartyexample1.md

```diff
@@ -0,0 +1,210 @@
+def startCapture():
+  pass
+
+# Stop capturing WebCam
+def stopCapture():
+  pass
+
+```
+{{</highlight>}}
+
+### Use OpenCV
+Your View2D is still empty, lets get access to the webcam and show the video in your module. Open the Python file of your network again and enter the following code:
+{{< highlight filename="<YOUR_MODULE_NAME>.py" >}}
+```Python
+# from mevis import *
+import cv2
+import OpenCVUtils
+
+_interfaces = []
+camera = None
+
+# Setup the interface for PythonImage module
+def setupInterface():
+  global _interfaces
+  _interfaces = []
+  interface = ctx.module("PythonImage").call("getInterface")
+  _interfaces.append(interface)
+
+# Release camera in the end
+def releaseCamera(_):
+  pass
+
+# Start capturing WebCam
+def startCapture():
+  pass
+
+# Stop capturing WebCam
+def stopCapture():
+  pass
+```
+{{</highlight>}}
+
+We now imported *cv2* and *OpenCVUtils* so that we can use them in Python. Additionally we defined list of *_interfaces* and a *camera*. The import of *mevis* is not necessary for this example.
+
+The *setupInterfaces* function is called whenever the *Window* of your module is opened. Here we are getting the interface of the `PythonImage` module and append it to our global list.
+
+### Access the Webcam
+Now we want to start capturing the camera.
+{{< highlight filename="<YOUR_MODULE_NAME>.py" >}}
+```Python
+# Start capturing WebCam
+def startCapture():
+  global camera
+  if not camera:
+    camera = cv2.VideoCapture(0)
+  ctx.callWithInterval(0.1, grabImage)
+
+# Grab image from camera and update
+def grabImage():
+  _, img = camera.read()
+  updateImage(img)
+
+# Update image in interface
+def updateImage(image):
+  _interfaces[0].setImage(OpenCVUtils.convertImageToML(image), minMaxValues = [0,255])
+```
+{{</highlight>}}
+
+The *startCapture* function gets the camera from the *cv2* object if not already available. Then it calls the current MeVisLab network context and creates a timer which calls a *grabImage* function every 0.1 seconds.
+
+The *grabImage* function reads an image from the *camera* and calls *updateImage*. The interface from the `PythonImage` module is used to set the image from the webcam. The MeVisLab *OpenCVUtils* convert the OpenCV image to the MeVisLab image format *MLImage*.
+
+Next, we define what happens if you click the *Pause* button.
+{{< highlight filename="<YOUR_MODULE_NAME>.py" >}}
+```Python
+...
+# Stop capturing WebCam
+def stopCapture():
+  ctx.removeTimers()
+...
+```
+{{</highlight>}}
+
+As we started a timer in our network context which updates the image every 0.1 seconds, we just stop this timer and the camera is paused.
+
+In the end, we need to release the camera whenever you close the Window of your Macro Module.
+{{< highlight filename="<YOUR_MODULE_NAME>.py" >}}
+```Python
+...
+# Release camera in the end
+def releaseCamera(_):
+  global camera, _interfaces
+  ctx.removeTimers()
+  _interfaces = []
+  if camera:
+    camera.release()
+    camera = None
```

Comment:
Shouldn't you do a `del camera` here?

Reply:
No, del camera deletes the object here which didn't work. Discussed with Uwe and this option works